client/src/components/profile/components/stats.tsx

```diff
@@ -0,0 +1,136 @@
+  const { t } = useTranslation();
+
+  /**
+   *  the following logic calculates streaks from the
+   *  users calendar
+   */
+
+  interface PageData {
+    startOfCalendar: Date;
+    endOfCalendar: Date;
+  }
+
+  interface CalendarData {
+    date: Date;
+    count: number;
+  }
+
+  // create array of timestamps and turn into milliseconds
+  const timestamps = Object.keys(calendar).map(
+    stamp => Number.parseInt(stamp, 10) * 1000
+  );
+  const startOfTimestamps = startOfDay(new Date(timestamps[0]));
+  let endOfCalendar = startOfDay(Date.now());
+  let startOfCalendar;
+
+  const pages: PageData[] = [];
+
+  do {
+    startOfCalendar = addDays(addMonths(endOfCalendar, -6), 1);
+
+    const newPage = {
+      startOfCalendar: startOfCalendar,
+      endOfCalendar: endOfCalendar
+    };
+
+    pages.push(newPage);
+
+    endOfCalendar = addDays(startOfCalendar, -1);
+  } while (startOfTimestamps < startOfCalendar);
+
+  pages.reverse();
+
+  const calendarData: CalendarData[] = [];
+  let dayCounter = pages[0].startOfCalendar;
+
+  // create an object for each day of the calendar period
+  while (dayCounter <= pages[pages.length - 1].endOfCalendar) {
+    const newDay = {
+      date: startOfDay(dayCounter),
+      count: 0
+    };
+
+    calendarData.push(newDay);
+    dayCounter = addDays(dayCounter, 1);
+  }
+
+  let longestStreak = 0;
+  let currentStreak = 0;
+  let lastIndex = -1;
+
+  // add a point to each day with a completed timestamp and calculate streaks
+  timestamps.forEach(stamp => {
+    const index = calendarData.findIndex(day =>
+      isEqual(day.date, startOfDay(stamp))
+    );
+
+    if (index >= 0) {
+      // add one point for today
+      calendarData[index].count++;
+
+      // if timestamp is on a new day, deal with streaks
+      if (index !== lastIndex) {
+        // if yesterday has points
+        if (calendarData[index - 1] && calendarData[index - 1].count > 0) {
+          currentStreak++;
+        } else {
+          currentStreak = 1;
+        }
+
+        if (currentStreak > longestStreak) {
+          longestStreak = currentStreak;
+        }
+      }
+
+      lastIndex = index;
+    }
+  });
+
+  // if today has no points
+  if (
+    calendarData[calendarData.length - 1] &&
+    calendarData[calendarData.length - 1].count === 0
+  ) {
+    currentStreak = 0;
+  }
+
+  return (
+    <FullWidthRow>
+      <h2 className='text-center'>Stats</h2>
+      <Spacer size='small' />
+      <div className='stats-container'>
+        <div>
+          <b data-testid='current-streak'>{t('profile.current-streak')}</b>
+          <p className='stat'>{currentStreak || 0}</p>
+        </div>
+        <div>
+          <b>{t('profile.total-points')}</b>
+          <p className='stat'>{points}</p>
+        </div>
+        <div>
+          <b data-testid='longest-streak'>{t('profile.longest-streak')}</b>
+          <p className='stat'>{longestStreak || 0}</p>
+        </div>
+      </div>
```

Comment:
This feels like it should have a little more meaning semantically, such as putting these in a description list.

Reply:
```html 
<div className='stats-container'>
  <dl>
    <dt><b data-testid='current-streak'>{t('profile.current-streak')}</b></dt>
    <dd className='stat'>{currentStreak || 0}</dd>

    <dt><b>{t('profile.total-points')}</b></dt>
    <dd className='stat'>{points}</dd>

    <dt><b data-testid='longest-streak'>{t('profile.longest-streak')}</b></dt>
    <dd className='stat'>{longestStreak || 0}</dd>
  </dl>
</div>
```
Something like this.